mailbox/jpa/src/main/java/org/apache/james/mailbox/jpa/mail/JPAMailboxMapper.java

```diff
@@ -82,9 +83,29 @@ protected void commit() throws MailboxException {
             throw new MailboxException("Commit of transaction failed", e);
         }
     }
+
+    @Override
+    public MailboxId create(Mailbox mailbox) throws MailboxException {
+        Preconditions.checkArgument(mailbox.getMailboxId() == null, "A mailbox we want to create should not have a mailboxId set already");
```

Comment:
I disagree implementing this. Isn't the default method in the API enough?

Reply:
You are right, I didn't cut it well, I will move the last commit to next PR, so we can adopt this one quickly

Reply:
-> #3108